udiskslinuxfilesystem: Sanitize label when creating the mount point

bertogg opened the pull request:
Here's a first attempt at fixing this problem. Apart from non-printable characters I also decided to remove a few other dangerous ones.

Ascii and non-ascii characters are treated separately as an optimization, but it's also possible to get rid of the ascii path altogether.

Fixes #1056

StorageGhoul commented:
Can one of the admins verify this patch?

bertogg commented:
I see that some tests failed but I cannot reproduce that problem in my environment, I assume the failures are unrelated to my patch, can anyone confirm that? Thanks!

vojtechtrefny commented:
> I see that some tests failed but I cannot reproduce that problem in my environment, I assume the failures are unrelated to my patch, can anyone confirm that? Thanks!

Yes, you can ignore the failures. These are known issues, mostly race conditions that happen more often on our slow CI machines.

vojtechtrefny reviewed (approved):
Looks good to me, thanks.

tbzatek reviewed (approved):
Looks good, thank you for your contribution.

Could you please add some tests? A while ago I added somewhat similar tests for fstab parser escaping - see #888

bertogg commented:
> Could you please add some tests? A while ago I added somewhat similar tests for fstab parser escaping - see #888

I just added some tests, tell me if they're ok or if you want me to change something.

Thanks!

bertogg commented on src/tests/dbus-tests/test_80_filesystem.py:
```diff
@@ -879,6 +879,72 @@ def test_protective_part_overwrite_mounted(self):
         part.Unmount(self.no_options, dbus_interface=self.iface_prefix + '.Filesystem')
         disk.Format(self._fs_signature, self.no_options, dbus_interface=self.iface_prefix + '.Block')
 
+    def _test_mountpoint(self, label):
```
On second thoughts it's probably better if this function receives the expected mount point as a parameter in addition to the label, I can update the pull request tomorrow.

bertogg replied:
In the end I decided to change it, and I added an additional test for utf-8 labels.

bertogg replied:
> Great, thanks! Good point about exfat and utf-8 labels.

FWIW this works fine in UDisks 2.9.4. This is because in _master_ the formatting is handled by libblockdev which sets `LC_ALL=C`: [here](https://github.com/storaged-project/libblockdev/blob/master/src/utils/exec.c#L215) and [here](https://github.com/storaged-project/libblockdev/blob/master/src/utils/exec.c#L419).

I can open a new issue about this.

bertogg replied:
Reported as https://github.com/storaged-project/libblockdev/issues/846

tbzatek replied:
Great, thanks! I was thinking about doing the same.
